routers/api/packages/cargo/cargo.go

```diff
@@ -48,7 +48,7 @@ func apiError(ctx *context.Context, status int, obj any) {
 
 // https://rust-lang.github.io/rfcs/2789-sparse-index.html
 func RepositoryConfig(ctx *context.Context) {
-	ctx.JSON(http.StatusOK, cargo_service.BuildConfig(ctx.Package.Owner))
+	ctx.JSON(http.StatusOK, cargo_service.BuildConfig(ctx.Package.Owner, ctx.Data["IsApiToken"] == true))
```

Comment:
I do not have the understanding for this package registry.

Just a guess:  `ctx.Package.Descriptor.Repository.IsPrivate` ?

Reply:
I think this might be a good idea, did it this way because the registry-auth tries the following on the sparse index:
GET {index_url}/config.json (without any credentials)
if 200:
  all subsquent requests will be unauthenticated
if 401:
  retry GET config.json with credentials
  if 200:
    if config.auth-required: include credentials for all requests
    else: include credentials only for publishing crates

Reply:
I implemented your change